### PR TITLE
Rename `db.operation` to `db.operation.name`

### PR DESCRIPTION
**Pull request by trask**
## Changes

Renames `db.operation` to `db.operation.name`, to be more future proof and allow other attributes under the `db.operation.*` namespace in the future.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: model/registry/db.yaml [hunk 1]
```diff
@@ -245,19 +245,16 @@ groups:
...
-          When setting this to an SQL keyword, it is not recommended to
-          attempt any client-side parsing of `db.statement` just to get this
-          property, but it should be set if the operation name is provided by
-          the library being instrumented.
-          If the SQL statement has an ambiguous operation, or performs more
-          than one operation, this value may be omitted.
+        requirement_level:
+          conditionally_required: >
+            If readily available. Otherwise, when the instrumentation library supports parsing `db.statement` and it's enabled,
+            `db.operation.name` SHOULD be set to the first table name found in the CQL query.
```
Comment (trask) [inline model/registry/db.yaml:257]:
```suggestion
            `db.operation.name` SHOULD be set to the first operation name found in the query.
```

**Comment by pyohannes** [comment]
In messaging, we have [`messaging.operation`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#messaging-attributes), and I wonder if consistency is important here.

Do you have an example of what additional attributes might be expected under `db.operation`?

**Comment by lmolkova** [comment]
> In messaging, we have [`messaging.operation`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#messaging-attributes), and I wonder if consistency is important here.
> 
> Do you have an example of what additional attributes might be expected under `db.operation`?

I think we might entertain the idea of `db.operation.duration` metric (and it might still be an open question for messaging metrics since operation list is an open set).

In case of databases, we have a precedent when [CosmosDB](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/cosmosdb.md#semantic-conventions-for-microsoft-cosmos-db) records operation name (e.g. `ReadItems`)  along with `db.cosmosdb.operation_type` (e.g. `Read`). 
If we translated it to messaging, we'd do something like `messaging.operation.type = settle, messaging.operation.name = ack`.

**Comment by lmolkova** [comment]
Also, graphql has `operation.name` and `operation.type` - https://github.com/open-telemetry/semantic-conventions/blob/main/docs/graphql/graphql-spans.md

I'm starting to understand why graphql used to be in the DB semconv - both try to describe QL

File: model/trace/database.yaml [hunk 2]
```diff
@@ -205,6 +207,8 @@ groups:
     brief: >
       Attributes for SQL databases
     attributes:
+      - ref: db.operation.name
+        examples: ['SELECT', 'INSERT', 'UPDATE', 'DELETE', 'CREATE', 'mystoredproc']
```
[the remark this answers is not in the document]
Comment (trask) [reply model/trace/database.yaml:215]: I've added a note to explicitly call out capturing stored procedure name for `EXEC` statements

**Review by lmolkova — APPROVED** [review]

File: model/registry/db.yaml [hunk 3]
```diff
@@ -245,19 +245,14 @@ groups:
...
         examples: [ 'customers', 'main' ]
         tag: db-generic
-      - id: operation
+      - id: operation.name
         type: string
         stability: experimental
         brief: >
-          The name of the operation being executed, e.g. the [MongoDB command name](https://docs.mongodb.com/manual/reference/command/#database-operations)
-          such as `findAndModify`, or the SQL keyword.
+          The name of the operation being executed.
```
Comment (pyohannes) [inline model/registry/db.yaml:252]:
```suggestion
          The name of the operation or command being executed.
```

Maybe? In strict SQL terminology, it's usually a "[command](https://www.w3schools.com/sql/sql_syntax.asp)" that's expected here (for example `SELECT`, or `INSERT`).

I also wonder if that would make the footnote obsolete, which seems to be a duplication of what's in the examples for this attribute.

**Comment by pyohannes** [comment]
Discussed in the messaging workgroup: for messaging we'll take over what's adopted for databases, as we don't see any strong reasons to stick with `messaging.operation` as an attribute (and be inconsistent with database semantic conventions).

See the related issue https://github.com/open-telemetry/semantic-conventions/issues/890.

**Review by jack-berg — APPROVED** [review]

**Review by gregkalapos — APPROVED** [review]

**Review by alanwest — APPROVED** [review]

**Pull request merged**